src/DynamoCore/Interfaces/IWorkspaceModel.cs

```diff
@@ -21,5 +21,11 @@ public interface IWorkspaceModel
         event Action NodesCleared;
         event Action<ConnectorModel> ConnectorAdded;
         event Action<ConnectorModel> ConnectorDeleted;
+
+        /// <summary>
+        /// Implement to record node modification for undo/redo
+        /// </summary>
+        /// <param name="nodeModel"></param>
+        void RecordModelForModification(ModelBase nodeModel);
```

Comment:
Don't forget to remove changes in `src/DynamoCore/Models/WorkspaceModel.cs` when this is removed.